ovh/resource_domain_name.go

```diff
@@ -0,0 +1,256 @@
+				fmt.Sprintf("Error calling Put %s", endpoint),
+				err.Error(),
+			)
+			return
+		}
+	}
+
+	// Read updated resource
+	if err := r.config.OVHClient.Get(endpoint, &responseData); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Get %s", endpoint),
+			err.Error(),
+		)
+		return
+	}
+
+	data.MergeWith(&responseData, true)
+
+	// Save data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *domainNameResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data, responseData DomainNameModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	endpoint := "/v2/domain/name/" + url.PathEscape(data.DomainName.ValueString())
+	if err := r.config.OVHClient.Get(endpoint, &responseData); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Get %s", endpoint),
+			err.Error(),
+		)
+		return
+	}
+
+	data.MergeWith(&responseData, true)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *domainNameResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data, planData, responseData DomainNameModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &planData)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	endpoint := "/v2/domain/name/" + url.PathEscape(data.DomainName.ValueString())
+
+	if !planData.TargetSpec.IsNull() && !planData.TargetSpec.IsUnknown() {
+		// Set the checksum to its current value
+		updateData := planData.ToUpdate()
+		updateData.Checksum = &data.Checksum
+
+		// Update resource
+		if err := r.config.OVHClient.Put(endpoint, updateData, nil); err != nil {
+			resp.Diagnostics.AddError(
+				fmt.Sprintf("Error calling Put %s", endpoint),
+				err.Error(),
+			)
+			return
+		}
+	}
+
+	// Read updated resource
+	if err := r.config.OVHClient.Get(endpoint, &responseData); err != nil {
+		resp.Diagnostics.AddError(
+			fmt.Sprintf("Error calling Get %s", endpoint),
+			err.Error(),
+		)
+		return
+	}
+
+	responseData.MergeWith(&planData, false)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &responseData)...)
+}
+
+func (r *domainNameResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data DomainNameModel
+
+	// Read Terraform prior state data into the model
+	resp.Diagnostics.Append(req.State.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	serviceName := data.DomainName.ValueString()
+
+	serviceID, err := serviceIdFromResourceName(r.config.OVHClient, serviceName)
```

Comment:
/services service name are not unique, meaning that domain name will have the same service name than the web hosting attached to the domain name, or the dns zone

As here https://github.com/ovh/terraform-provider-ovh/blob/257f28bc03118e56bfead285a616cb7c8234dda8/ovh/services.go#L57, we are returning the first service, we might not delete the right one.

We need to make `serviceIdFromResourceName` evolve to handle "serviceType"